flink-table/flink-table-api-scala/src/main/scala/org/apache/flink/table/api/ImplicitExpressionConversions.scala

```diff
@@ -726,6 +726,25 @@ trait ImplicitExpressionConversions {
     Expressions.ifThenElse(condition, ifTrue, ifFalse)
   }
 
+  /**
+   * Returns the first argument that is not NULL.
+   *
+   * If all arguments are NULL, it returns NULL as well. The return type is the least
+   * restrictive, common type of all of its arguments. The return type is nullable if all
+   * arguments are nullable as well.
+   *
+   * For example:
+   *
+   *  - `coalesce(null, default")` returns `"default"`.
+   *  - `coalesce($"f0", $"f1", "default")` returns the first non-null value among f0 and f1,
+   *    or `"default"` if f0 and f1 are both null
```

Comment:
This isn't yet a code block like with Java. There's also a syntax error in there.

```suggestion
   * Examples:
   * {{{
   * // Returns "default"
   * coalesce(null, "default")
   *
   * // Returns the first non-null value among f0 and f1, or "default" if f0 and f1 are both null
   * coalesce($"f0", $"f1", "default")
   * }}}
```